From 15afa0b9293e986eda61f0690a1a9349dacdee83 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Thu, 17 Oct 2019 14:32:42 +0300
Subject: [PATCH 02/39] <sys/uio.h>: Cleanup namespace.

 Certain care is needed while handling this, lots of namespace issues.
 The <sys/_null.h> is needed for sys/vfs/hammer/hammer_ondisk.c through
 <sys/nlookup.h> that would benefit from separated uio enums.
---
 sys/sys/uio.h | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/sys/sys/uio.h b/sys/sys/uio.h
index aca9068b8e..bee1298c85 100644
--- a/sys/sys/uio.h
+++ b/sys/sys/uio.h
@@ -35,13 +35,20 @@
 
 #include <sys/cdefs.h>
 #include <sys/_iovec.h>
-#if __BSD_VISIBLE
-#include <sys/param.h>
-#endif
 #if defined(_KERNEL)
+#include <sys/_null.h>		/* XXX */
 #include <sys/malloc.h>		/* Needed to inline iovec_free(). */
 #endif
 
+#ifdef __BSD_VISIBLE
+#ifndef _OFF_T_DECLARED
+typedef	__off_t		off_t;
+#define	_OFF_T_DECLARED
+#endif
+#endif
+
+/* The size_t and __ types are expected to be provided by <sys/_iovec.h> */
+
 #ifndef _SSIZE_T_DECLARED
 typedef	__ssize_t	ssize_t;
 #define	_SSIZE_T_DECLARED
@@ -69,8 +76,6 @@ enum uio_seg {
  *	 iov uses an unsigned quantity, DragonFly will use the (unsigned)
  *	 size_t.
  */
-struct buf;
-
 struct uio {
 	struct	iovec *uio_iov;
 	int	uio_iovcnt;
@@ -87,10 +92,10 @@ struct uio {
 #define UIO_MAXIOV	1024		/* max 1K of iov's */
 #define UIO_SMALLIOV	8		/* 8 on stack, else malloc */
 
-#endif
+#endif /* _KERNEL || _KERNEL_STRUCTURES */
 
 #if defined(_KERNEL)
-
+struct buf;
 struct vm_object;
 struct vm_page;
 
@@ -116,9 +121,9 @@ iovec_free(struct iovec **kiov, struct iovec *siov)
 		*kiov = NULL;
 	}
 }
+#endif /* _KERNEL */
 
-#else /* !_KERNEL */
-
+#ifndef _KERNEL
 __BEGIN_DECLS
 ssize_t	readv(int, const struct iovec *, int);
 ssize_t	writev(int, const struct iovec *, int);
@@ -127,7 +132,6 @@ ssize_t	preadv(int, const struct iovec *, int, off_t);
 ssize_t	pwritev(int, const struct iovec *, int, off_t);
 #endif
 __END_DECLS
-
-#endif /* _KERNEL */
+#endif /* !_KERNEL */
 
 #endif /* !_SYS_UIO_H_ */
-- 
2.23.0

